packages/babel-plugin-proposal-optional-chaining/src/index.js

```diff
@@ -116,14 +128,7 @@ export default declare((api, options) => {
           }
           let replacement = replacementPath.node;
           // Ensure (a?.b)() has proper `this`
-          if (
-            t.isMemberExpression(replacement) &&
-            (replacement.extra?.parenthesized ||
-              // if replacementPath.parentPath does not equal parentPath,
-              // it must be unwrapped from parenthesized expression.
-              replacementPath.parentPath !== parentPath) &&
-            parentPath.isCallExpression()
-          ) {
+          if (i === 0 && parentIsCall) {
```

Comment:
What happens if we remove the `i` check?

Reply:
Good question!

Since `parentIsCall` is computed only once for the `baseRef`, if we don't limit this branch with `i === 0`, which means applying to the baseRef only, it will generate multiple memoizing assignment during looping all the optional chain elements.

This change fixes an issue in https://github.com/babel/babel/pull/11662/files#diff-778d6015479bf103795200e754865e9dL30 where `fn().Foo.self` is memoized twice.

Reply:
Do we want to add this in a comment?